ProjFS.Mac/PrjFSKext/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -124,6 +124,19 @@ kern_return_t VirtualizationRoots_Init()
 
 kern_return_t VirtualizationRoots_Cleanup()
 {
+    if (s_virtualizationRoots != nullptr)
+    {
+        for (uint32_t i = 0; i < s_maxVirtualizationRoots; ++i)
+        {
+            // If there are still providers registered at this point, we will leak vnodes
```

Comment:
For my own understanding, does `vfs_unmountall` block until all vnodes are released? (I am assuming this was the cause of the hang?)

Reply:
I'd have to read the source code to say for sure, but I assume the answer is yes because vnodes have data attached to them that's private to the file system to which they belong. So it's potentially not safe to keep vnodes around after their file system has been unmounted (and the file system's kext potentially unloaded); with that background it makes sense to me that allowing an unmount to complete would require all associated vnodes to be recycled.